interface/super/edit_layout.php

```diff
@@ -1969,28 +1925,22 @@ function validateNewField(idpfx) {
         }
         var validname = $("#newgroupname").val().replace(/[^A-za-z0-9 ]/g, "_"); // match any non-word characters and replace them
         $("#newgroupname").val(validname);
-
-        // now, check the first group field values
-        if (!validateNewField('gnew')) return false;
-
-        // submit the form to add a new field to a specific group
         $("#formaction").val("addgroup");
         mySubmit();
     }
 
-    /******************************************************************
     // actually delete an entire group from the database
     var DeleteGroup = function(btnObj) {
         var parts = $(btnObj).attr("id");
-        var groupname = parts.replace(/^\d+/, "");
-        if (confirm(<?php echo xlj('WARNING') ?> + " - " + <?php echo xlj('This action cannot be undone.') ?> + "\n" + <?php echo xlj('Are you sure you wish to delete the entire group named'); ?> + " '" + groupname + "'?")) {
+        if (confirm("<?php echo xls('WARNING') . ' - ' . xls('This action cannot be undone.') . '\n' .
+            xls('Are you sure you wish to delete this entire group?'); ?>")
```

Comment:
should be using xlj instead of xls

Reply:
Hmm. xls() is part of a javascript string literal, so json_encode() output there will not be evaluated as an expression. Do you mean like this:

        if (confirm(<?php echo js_escape(xl('WARNING') . ' - ' . xl('This action cannot be undone.') . '\n' .
            xl('Are you sure you wish to delete this entire group?')); ?>)

... but I wonder what js_escape() will do to my '\n' which is intended to force a line break.

Reply:
Guessing that could cause issues. Aren't able to separate out the \n character like is done in above line 1986 that was removed?

Reply:
Yep I'll try that.